decentraland/unity-explorer
# test: memory investigation

## What does this PR change?

Memory optimizations

- Reduced initial size of TextureArray2D, both for Avatars and LODs
- Added a throttled `Resources.UnloadUnusued` on `ReleaseMemory` system
- Changed the system memory used metric to `SystemUsedMemoryInBytes` since it reflects better the current memory usage
- Reduce the memory limits for unloading cached
- Added the possibility to take load radius to 1 for testing purposes


Loading Screen

- Remove loading screen if we are in memory warning, since assets may not load therefore falling into a deadlock. We need a popup
- Added timeout for `roomHub.StopIfNotAsync()` and `roomHub.StartAsync()` since it was holding some realm changes. 

Debug

- Adds `Web Request Deadlock` Info
- Adds `Loading Screen Step` Info

## How to test the changes?

1. Launch the explorer in MAC
2. In your settings, lets set the radius to 20.
3. Play normally. Teleport, change realms, run. Whatever comes to mind
4. If you get stuck in the loading screen, please take a screenshot of the first widgets in the debug panel. This three:
<img width="244" alt="image" src="https://github.com/user-attachments/assets/3ca4b909-413e-49e2-9187-c6b0d0b406e9">

5. Check particulary the `Request cannot connect` field. Check if that number keeps increasing while stuck
6. If stuck, please take a screenshot of the `Activity Monitor` application and attach it in your report. I would recommend it having open from start, open Decentraland and Activity Monitor together. It looks like this:

![image](https://github.com/user-attachments/assets/e019a2fc-a21a-4658-bc01-b5ae247de90a)

7. Repeat a couple of times. Get the log for each frozen run

--------------

After completing the steps above, lets do the same test but with an extra manual step

Repeat all the steps mentioned above, but before each teleport, press the FULL button, wait for `TOTAL USED MEMORY` field  to stabilize, then press NORMAL
, and then teleport

<img width="301" alt="image" src="https://github.com/user-attachments/assets/1b853196-6478-4cc0-967f-1a9eb5e14e80">


## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
